sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -465,6 +465,8 @@ case class DataSource(
     providingClass.newInstance() match {
       case dataSource: CreatableRelationProvider =>
         SaveIntoDataSourceCommand(data, dataSource, caseInsensitiveOptions, mode)
+      case dataSource: ConsoleSinkProvider =>
+        data.show(data.count().toInt, false)
```

Comment:
Sorry for late reply.
Yes, it is right to use underscore since dataSource is not used. Considering it is no need to create a new ConsoleSink and no access to the private variable, i will use caseInsensitiveOptions instead to extract the numRows and truncate, 
Thanks for your comment.